src/StudentsController.ts

```diff
@@ -165,7 +169,36 @@ export class StudentsController {
         await this.#studentsCollection.update(oldDoc, student.toJSON());
     }
 
+    public async pseudonymizeStudent(student: Student): Promise<Student> {
+        // Only keep id and status of student (also works for future student properties)
+        const oldDoc = await this.#studentsCollection.read(student.id.toString());
+        const pseudonymizedDoc = {
+            id: oldDoc.id,
+            status: "deleted"
```

Comment:
I don't see a `status` property in the `Student` class. Also: the property should probably rather be an enum instead of a string. Or at least a union type (`"deleted"|...`)